nx.json

```diff
@@ -3,52 +3,52 @@
     "default": {
       "runner": "nx/tasks-runners/default",
       "options": {
-        "cacheableOperations": []
+        "cacheableOperations": ["cached:build:stencil"]
       }
     }
   },
+  "namedInputs": {
+    "sharedGlobals": [{"runtime": "node -v"}],
+    "sharedTypescriptSources": [
+      "{workspaceRoot}/tsconfig.json",
+      "{projectRoot}/tsconfig.json",
+      "{projectRoot}/src/**",
+      "!{projectRoot}/src/**/?(*.)+({{spec.{ts,tsx},spec.{ts,tsx}.snap},{?(*([^.]).)stories.{tsx,css},mdx}})"
+    ],
+    "production": [
+      "sharedGlobals",
+      "{projectRoot}/**",
+      "!{projectRoot}/**/?(*.)+({{spec.{ts,tsx},spec.{ts,tsx}.snap},{?(*([^.]).)stories.{tsx,css},mdx}})",
+      "!{projectRoot}/{dist,www,temp}/**"
+    ]
+  },
   "targetDefaults": {
     "build": {
-      "dependsOn": [
-        "^build"
-      ]
+      "inputs": ["^production", "production"],
+      "dependsOn": ["^build"]
     },
     "prod": {
-      "dependsOn": [
-        "build"
-      ]
+      "dependsOn": ["build"]
     },
     "test": {
-      "dependsOn": [
-        "^build"
-      ]
+      "dependsOn": ["^build"]
     },
     "e2e": {
-      "dependsOn": [
-        "build"
-      ]
+      "dependsOn": ["build"]
     },
     "lint:check": {},
     "lint:fix": {},
     "publish:npm:release": {
-      "dependsOn": [
-        "build"
-      ]
+      "dependsOn": ["build"]
```

Comment:
- [ ] ensure nx.json is prettified
- [ ] ensure nx.json is protected by the lint step on CI

Reply:
can be split in another PR, but still, that's a req.

Reply:
our lint step uses ESLint, which doesn't work with JSON without a plugin. I prefer to stick to just using Prettier, so I'll just add Prettier to the lint-staged rules.